File: docs/_src/wpf/FileSystemBrowserDocs.cs
using System;
using Xarial.XToolkit;
using Xarial.XToolkit.Wpf.Services;
using Xarial.XToolkit.Services;

namespace Wpf.Docs
{
    public static class FileSystemBrowserDocs
    {
        public static void BrowseFolder() 
        {
            //--- browse-folder
            var fsb = new FileSystemBrowser();

            if (fsb.BrowseFolder(out string path, "Select sample folder"))
            {
                Console.WriteLine($"Selected path: {path}");
            }
            else 
            {
                Console.WriteLine("User has cancelled the folder browsing");
            }
            //---
        }

        public static void BrowseFile() 
        {
            //--- browse-file-filter
            var filters = FileFilter.BuildFilterString(
                new FileFilter("Text Files", "*.txt", "*.doc", "*.md"), 
                FileFilter.ImageFiles, 
                FileFilter.AllFiles);
            //---

            //--- browse-file-open
            var fsb = new FileSystemBrowser();

            if (fsb.BrowseFileOpen(out string fileIn, "Select input file", filters)) 
            {
                Console.WriteLine($"Selected path: {fileIn}");
            }
            else
            {
                Console.WriteLine("User has cancelled the file browsing");
            }
            //---

            //--- browse-file-save
            if (fsb.BrowseFileSave(out string fileOut, "Select output file", filters))
            {
                Console.WriteLine($"Selected path: {fileOut}");
            }
            else
            {
                Console.WriteLine("User has cancelled the file browsing");
            }
            //---
        }
    }
}

File: docs/_src/wpf/FileSystemBrowserDocs.vb
Imports System
Imports Xarial.XToolkit
Imports Xarial.XToolkit.Wpf.Utils

Namespace Wpf.Docs

    Module FileSystemBrowserDocs

        Sub BrowseFolder()

            '--- browse-folder
            Dim path As String = Nothing

            If FileSystemBrowser.BrowseFolder(path, "Select sample folder") Then
                Console.WriteLine($"Selected path: {path}")
            Else
                Console.WriteLine("User has cancelled the folder browsing")
            End If
            '---

        End Sub

        Sub BrowseFile()

            '--- browse-file-filter
            Dim filters = FileFilter.BuildFilterString(
                New FileFilter("Text Files", "*.txt", "*.doc", "*.md"),
                FileFilter.ImageFiles,
                FileFilter.AllFiles)
            '---

            '--- browse-file-open
            Dim fileIn As String = Nothing

            If FileSystemBrowser.BrowseFileOpen(fileIn, "Select input file", filters) Then
                Console.WriteLine($"Selected path: {fileIn}")
            Else
                Console.WriteLine("User has cancelled the file browsing")
            End If
            '---

            '--- browse-file-save
            Dim fileOut As String = Nothing

            If FileSystemBrowser.BrowseFileSave(fileOut, "Select output file", filters) Then
                Console.WriteLine($"Selected path: {fileOut}")
            Else
                Console.WriteLine("User has cancelled the file browsing")
            End If
            '---

        End Sub

    End Module

End Namespace
